### PR TITLE
474/soroban diagnostic events

### PR DESCRIPTION
**Pull request by paulbellamy**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Add support for captive-core to emit Soroban diagnostic events

### Why

Part of https://github.com/stellar/soroban-tools/issues/474

- For the captive-core config, soroban-rpc needs to be able to enable this on captive-core.
- ~For horizon, I just added the flag for completeness. Not super committed to it, as we don't do anything with the events yet anyway (afaik?)~

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: services/horizon/internal/flags.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ const (
 	CaptiveCoreConfigPathName = "captive-core-config-path"
 	// captive-core-use-db is the command line flag for enabling captive core runtime to use an external db url connection rather than RAM for ledger states
 	CaptiveCoreConfigUseDB = "captive-core-use-db"
+	// captive-core-enable-soroban-diagnostic-events is the command line flag for enabling soroban diagnostic events in captive core
+	CaptiveCoreConfigEnableSorobanDiagnosticEvents = "captive-core-enable-soroban-diagnostic-events"
```
Comment (tsachiherman) [inline services/horizon/internal/flags.go:44]: This is not a bug.. but I can't think of any reason we would need this. Am I missing something ?

**Review by tsachiherman — COMMENTED** [review]
I could be wrong, but I don't believe that horizon would need to support this flag.
This flag would only be used in the soroban-rpc.

File: services/horizon/internal/config.go [hunk 2]
```diff
@@ -19,17 +19,18 @@ type Config struct {
...
+	UsingDefaultPubnetConfig                       bool
+	CaptiveCoreBinaryPath                          string
+	RemoteCaptiveCoreURL                           string
+	CaptiveCoreConfigPath                          string
+	CaptiveCoreTomlParams                          ledgerbackend.CaptiveCoreTomlParams
+	CaptiveCoreToml                                *ledgerbackend.CaptiveCoreToml
+	CaptiveCoreStoragePath                         string
+	CaptiveCoreReuseStoragePath                    bool
+	CaptiveCoreConfigUseDB                         bool
+	CaptiveCoreConfigEnableSorobanDiagnosticEvents bool
```
Comment (Shaptic) [inline services/horizon/internal/config.go:33]: Wasn't the point of introducing the captive core toml file (#3442) to reduce the need for adding incredibly specific flags that configure captive core's behavior? We can probably just get away with leaving it there

File: services/horizon/internal/flags.go (hunk 1 above)
Comment (paulbellamy) [reply services/horizon/internal/flags.go:44]: Horizon shouldn't need it no. Just included it for completeness. Happy to remove if people feel that's better.

File: services/horizon/internal/config.go (hunk 2 above)
Comment (paulbellamy) [reply services/horizon/internal/config.go:33]: You mean to horizon? or generally? Like are you suggesting we should just always enable diagnostic events with captive core?

Comment (Shaptic) [reply services/horizon/internal/config.go]: I'm saying that Horizon should parse whatever Captive Core gives it, and people should use the toml file to control whether or not CC includes diagnostic events.

File: services/horizon/internal/flags.go (hunk 1 above)
Comment (paulbellamy) [reply services/horizon/internal/flags.go]: Good point. Removed.

File: services/horizon/internal/config.go (hunk 2 above)
Comment (paulbellamy) [reply services/horizon/internal/config.go]: Agreed. Removed.

**Review by tsachiherman — APPROVED** [review]

**Review by Shaptic — APPROVED** [review]
:shipit:

**Pull request merged**